src/net/cooperi/pivapplet/PivApplet.java

```diff
@@ -2417,8 +2417,10 @@ else if (key == (byte)0x81 && pukPinIsDefault)
 				return;
 
 			} else {
-				ISOException.throwIt((short)(
-				    (short)0x63C0 | pin.getTriesRemaining()));
+				short tries = pin.getTriesRemaining();
+				if (tries > 0x0f)
+					tries = 0x0f;
+				ISOException.throwIt((short) ((short) 0x63C0 | tries));
```

Comment:
Let's put that into a reusable method

Reply:
I created a method `throwPinVerificationException`